File: cs50x projects/credit/credit.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>

bool checkSum(char *a);
int checkCredit(char *a);
bool checkInput(char *a);

int main(void)
{
    string a;

    do{
       a =get_string("please enter the number");
    } while(!checkInput(a));

    if(checkSum(a))
    {
        if(checkCredit(a)==0) {printf("AMEX\n"); return 0; }
        if(checkCredit(a)==1) {printf("MASTERCARD\n"); return 0;}
        if(checkCredit(a)==2) {printf("VISA\n"); return 0;}
    }
    printf("INVALID\n");
    return 0;

}

bool checkSum(char *a){

    int len=strlen(a);
    int sum=0;
    int sum2=0;
     for(int i=len-2; i>=0; i-=2)
    {
        int temp=(a[i]-'0')*2;
        while(temp%10!=temp)
        {
            int n=temp%10;
            sum+=n;
            temp/=10;

        }
        sum+=temp;
    }

     for(int j=len-1; j>=0; j-=2)
    {
        sum2+=a[j]-'0';
    }

        int b=sum+sum2;
        if(b%10==0)
        {
            return true;
        }
        else return false;
}

int checkCredit(char *a)
{
    int len=strlen(a);

    if(((a[0]-'0'==3&&(a[1])-'0'==7&&len==15)||
    ((a[0])-'0'==3&&(a[1])-'0'==4))&&len==15)
    {
        return 0;
    }

    if((a[0]-'0'==5&&a[1]-'0'==1&&len==16)||
    ((a[0]-'0'==5&&a[1]-'0'==2&&len==16))||
    ((a[0]-'0'==5&&a[1]-'0'==3&&len==16))||
    ((a[0]-'0'==5&&a[1]-'0'==4&&len==16))||
    ((a[0]-'0'==5&&a[1]-'0'==5&&len==16)))
    {
        return 1;
    }

    if( (((a[0])-'0'==4)&& len==13)||(((a[0])-'0'==4)&&len==16) )
    {
        return 2;
    }

    return 400;
}

bool checkInput(char *a)
{
    int len=strlen(a);
    for(int i=0; i<len;i++)
    {
        if(a[i]<48 || a[i]>57)
        {
            return false;
        }
    }
     return true;
}

File: cs50x projects/sentimental-credit/credit.py

from cs50 import get_int


def main():
    # Prompts user for card number
    number = get_int("Number: ")

    # If card is valid, then check the card company
    if check_valid(number):
        card_Type(number)
    # Card not valid
    else:
        print("INVALID")


def check_valid(num):
    # Calculates sum according to Luhn's Algorithm
    sum = 0
    for i, c in enumerate(reversed(str(num))):
        if i % 2 == 0:
            sum += int(c)
        else:
            for j in str(int(c) * 2):
                sum += int(j)

    # Checks if sum is divisible by 10
    if sum % 10 == 0:
        return True
    else:
        return False


def card_Type(card):
    # Removes all the digits from card number except first 2
    num = int(str(card)[0:2])

    # Check for conditions of all companies
    if (num == 34 or num == 37) and len(str(card)) == 15:
        print("AMEX")
    elif num > 50 and num < 56 and len(str(card)) == 16:
        print("MASTERCARD")
    elif num >= 40 and num < 50 and (len(str(card)) == 13 or len(str(card)) == 16):
        print("VISA")
    else:
        print("INVALID")


main()




# def main():

#     num=input("please enter the number")

#     while checkInput(num)==False:

#             num=input("please enter correct nuber")


#     if checkSum(num):
#         if checkCredit(num)==0:
#             print("AMEX")
#         if checkCredit(num)==1:
#             print("MASTERCARD")
#         if checkCredit(num)==2:
#             print("VISA")

#     print("INVALID")


# def checkSum(num):
#     le=len(num)
#     sum=0
#     sum2=0

#     for i in range(le-2,0,-2):
#         temp=(int(num[i]))*2
#         while(temp%10!=temp):
#             n=temp%10
#             sum+=n
#             temp/=10

#     for j in range(le-1,0,-2):
#         sum2+=int(num[j])

#     result=sum+sum2
#     if(result%10==0):
#         return True
#     else:
#          return False

# def checkCredit(num):
#     le=len(num)

#     if int(num[0])==3 and int(num[1])==7 and le==15:
#         if int(num[0])==3 and int(num[1])==4 and le==15:
#            return 0

#     if (int(num[0])==5 and int(num[1])==1 and le==16):
#         if(int(num[0])==5 and int(num[1])==2 and le==16):
#            if(int(num[0])==5 and int(num[1])==4 and le==16):
#                if(int(num[0])==5 and int(num[1])==5 and le==16):
#                      return 1

#     if (int(num[0])==4 and le==13):
#         if(int(num[0])==4 and le==16):
#             return 2
#     return 400




# def checkInput(num):
#  c=False
#  for i in range(len(num)):
#          if num[i].isdigit()==True:
#              c=True
#          else:
#              c= False
#              return c
#  return c



# main()


